Allow @types/prop-types to be ^15.5.3, not exactly 15.5.3?

Here's a snippet from my `yarn.lock`:

```yaml
office-ui-fabric-react@6.52.0:
  version "6.52.0"
  resolved "https://office.pkgs.visualstudio.com/_packaging/Sway/npm/registry/office-ui-fabric-react/-/office-ui-fabric-react-6.52.0.tgz#7a68f85a316296a1b61e209a42a2ed0d63483361"
  dependencies:
    "@microsoft/load-themed-styles" "^1.7.13"
    "@types/prop-types" "15.5.3"
    "@uifabric/icons" ">=6.1.0 <7.0.0"
    "@uifabric/merge-styles" ">=6.7.0 <7.0.0"
    "@uifabric/styling" ">=6.17.0 <7.0.0"
    "@uifabric/utilities" ">=6.14.1 <7.0.0"
    prop-types "^15.5.10"
    tslib "^1.7.1"
```

Because `@types/prop-types` is pinned to exactly 15.5.3, my `@types/react` seems to be refusing to resolve to exactly one `*` _(despite every package allowing `*` all versions)_:

```yaml

"@types/react@*":
  version "16.4.11"
  resolved "https://office.pkgs.visualstudio.com/_packaging/Sway/npm/registry/@types/react/-/react-16.4.11.tgz#330f3d864300f71150dc2d125e48644c098f8770"
  dependencies:
    "@types/prop-types" "*"
    csstype "^2.2.0"

"@types/react@16.4.0":
  version "16.4.0"
  resolved "https://office.pkgs.visualstudio.com/_packaging/Sway/npm/registry/@types/react/-/react-16.4.0.tgz#f7f639837bf1f8635f57616b12a1b0d193b6343a"
  dependencies:
    csstype "^2.2.0"
```

As a result, `tsc` is blowing up with duplicate definition complaints for JSX/React types:

```
node_modules/@types/enzyme/node_modules/@types/react/index.d.ts:2305:19 - error TS2320: Interface 'ElementClass' cannot simultaneously extend types 'Component<any, {}, any>' and 'Component<any, {}, any>'.
  Named property 'props' of types 'Component<any, {}, any>' and 'Component<any, {}, any>' are not identical.

2305         interface ElementClass extends React.Component<any> {
```

Is there a reason the types need to be _exactly_ `15.5.3`? The result is that every other package gets the same, updated `@types/prop-types` except for `office-ui-fabric-react`:

```yarml
"@types/prop-types@*", "@types/prop-types@15.5.5", "@types/prop-types@^15.5.2":
  version "15.5.5"
  resolved "https://office.pkgs.visualstudio.com/_packaging/Sway/npm/registry/@types/prop-types/-/prop-types-15.5.5.tgz#17038dd322c2325f5da650a94d5f9974943625e3"
  dependencies:
    "@types/react" "*"

"@types/prop-types@15.5.3":
  version "15.5.3"
  resolved "https://office.pkgs.visualstudio.com/_packaging/Sway/npm/registry/@types/prop-types/-/prop-types-15.5.3.tgz#bef071852dca2a2dbb65fecdb7bfb30cedae2de2"
```

--- a/apps/fabric-website-resources/package.json
+++ b/apps/fabric-website-resources/package.json
@@ -54,7 +54,7 @@
   },
   "dependencies": {
     "@microsoft/load-themed-styles": "^1.7.13",
-    "@types/prop-types": "15.5.3",
+    "@types/prop-types": "^15.5.3",
     "@uifabric/example-app-base": ">=6.7.2 <7.0.0",
     "@uifabric/icons": ">=6.1.0 <7.0.0",
     "@uifabric/merge-styles": ">=6.7.0 <7.0.0",

--- a/apps/fabric-website-resources/package.json
+++ b/apps/fabric-website-resources/package.json
@@ -28,6 +28,7 @@
     "@types/enzyme-adapter-react-16": "1.0.3",
     "@types/es6-promise": "0.0.32",
     "@types/jest": "23.0.0",
+    "@types/prop-types": "^15.5.3",
     "@types/react": "16.3.16",
     "@types/react-dom": "16.0.5",
     "@types/react-test-renderer": "^16.0.0",
@@ -54,7 +55,6 @@
   },
   "dependencies": {
     "@microsoft/load-themed-styles": "^1.7.13",
-    "@types/prop-types": "^15.5.3",
     "@uifabric/example-app-base": ">=6.7.2 <7.0.0",
     "@uifabric/icons": ">=6.1.0 <7.0.0",
     "@uifabric/merge-styles": ">=6.7.0 <7.0.0",

--- a/common/changes/@uifabric/charting/master_2018-08-23-02-06.json
+++ b/common/changes/@uifabric/charting/master_2018-08-23-02-06.json
@@ -1,0 +1,11 @@
+{
+  "changes": [
+    {
+      "packageName": "@uifabric/charting",
+      "comment": "Fix @types/prop-types being inconsistently restrictive.",
+      "type": "patch"
+    }
+  ],
+  "packageName": "@uifabric/charting",
+  "email": "cliff.koh@microsoft.com"
+}

--- a/common/changes/@uifabric/fabric-website-resources/master_2018-08-23-02-06.json
+++ b/common/changes/@uifabric/fabric-website-resources/master_2018-08-23-02-06.json
@@ -1,0 +1,11 @@
+{
+  "changes": [
+    {
+      "packageName": "@uifabric/fabric-website-resources",
+      "comment": "Fix @types/prop-types being inconsistently restrictive.",
+      "type": "patch"
+    }
+  ],
+  "packageName": "@uifabric/fabric-website-resources",
+  "email": "cliff.koh@microsoft.com"
+}

--- a/common/changes/@uifabric/utilities/master_2018-08-23-02-06.json
+++ b/common/changes/@uifabric/utilities/master_2018-08-23-02-06.json
@@ -1,0 +1,11 @@
+{
+  "changes": [
+    {
+      "packageName": "@uifabric/utilities",
+      "comment": "Fix @types/prop-types being inconsistently restrictive.",
+      "type": "patch"
+    }
+  ],
+  "packageName": "@uifabric/utilities",
+  "email": "cliff.koh@microsoft.com"
+}

--- a/common/changes/office-ui-fabric-react/master_2018-08-23-02-06.json
+++ b/common/changes/office-ui-fabric-react/master_2018-08-23-02-06.json
@@ -1,0 +1,11 @@
+{
+  "changes": [
+    {
+      "packageName": "office-ui-fabric-react",
+      "comment": "Fix @types/prop-types being inconsistently restrictive.",
+      "type": "patch"
+    }
+  ],
+  "packageName": "office-ui-fabric-react",
+  "email": "cliff.koh@microsoft.com"
+}

--- a/packages/charting/package.json
+++ b/packages/charting/package.json
@@ -29,6 +29,7 @@
     "@types/jest": "23.0.0",
     "@types/enzyme": "3.1.13",
     "@types/enzyme-adapter-react-16": "1.0.3",
+    "@types/prop-types": "^15.5.3",
     "@types/webpack-env": "1.13.0",
     "@uifabric/example-app-base": ">=6.7.2 <7.0.0",
     "@uifabric/jest-serializer-merge-styles": ">=6.0.4 <7.0.0",

--- a/packages/office-ui-fabric-react/package.json
+++ b/packages/office-ui-fabric-react/package.json
@@ -29,6 +29,7 @@
     "@types/enzyme-adapter-react-16": "1.0.3",
     "@types/es6-promise": "0.0.32",
     "@types/jest": "23.0.0",
+    "@types/prop-types": "^15.5.3",
     "@types/react": "16.3.16",
     "@types/react-dom": "16.0.5",
     "@types/react-test-renderer": "^16.0.0",
@@ -55,7 +56,6 @@
   },
   "dependencies": {
     "@microsoft/load-themed-styles": "^1.7.13",
-    "@types/prop-types": "^15.5.3",
     "@uifabric/icons": ">=6.1.0 <7.0.0",
     "@uifabric/merge-styles": ">=6.7.0 <7.0.0",
     "@uifabric/styling": ">=6.17.0 <7.0.0",

--- a/packages/office-ui-fabric-react/package.json
+++ b/packages/office-ui-fabric-react/package.json
@@ -55,7 +55,7 @@
   },
   "dependencies": {
     "@microsoft/load-themed-styles": "^1.7.13",
-    "@types/prop-types": "15.5.3",
+    "@types/prop-types": "^15.5.3",
     "@uifabric/icons": ">=6.1.0 <7.0.0",
     "@uifabric/merge-styles": ">=6.7.0 <7.0.0",
     "@uifabric/styling": ">=6.17.0 <7.0.0",

--- a/packages/utilities/package.json
+++ b/packages/utilities/package.json
@@ -42,7 +42,7 @@
     "sinon": "^4.1.3"
   },
   "dependencies": {
-    "@types/prop-types": "^15.5.2",
+    "@types/prop-types": "^15.5.3",
     "@uifabric/merge-styles": ">=6.7.0 <7.0.0",
     "prop-types": "^15.5.10",
     "tslib": "^1.7.1"

--- a/packages/utilities/package.json
+++ b/packages/utilities/package.json
@@ -28,6 +28,7 @@
     "@types/enzyme": "3.1.13",
     "@types/enzyme-adapter-react-16": "1.0.3",
     "@types/jest": "23.0.0",
+    "@types/prop-types": "^15.5.3",
     "@types/react": "16.3.16",
     "@types/react-dom": "16.0.5",
     "@types/react-test-renderer": "^16.0.0",
@@ -42,7 +43,6 @@
     "sinon": "^4.1.3"
   },
   "dependencies": {
-    "@types/prop-types": "^15.5.3",
     "@uifabric/merge-styles": ">=6.7.0 <7.0.0",
     "prop-types": "^15.5.10",
     "tslib": "^1.7.1"

